### DISCUSSION TITLE
引用时上标和正常格式混用问题

### DISCUSSION THREAD

**Discussion by lsqzero** [zotero-chinese/wiki#271] [求助 / Q&A] [answered]
请问个问题。我是一名工科学生，我在Word撰写期刊文章，引用文章时，会出现上标和正常格式混用的情况。例如：
”文献[1]利用了XXX算法，提高了效率，然而准确度有所下降<sup>[2]</sup>“
目前样式似乎只能插入上标，因此我只能手动在word当中调整格式，但每次新插入文献刷新后，都需要重新调整。请问是否有更简便的解决方式，能够指定该插入的文献为上标或者正常格式。非常感谢！

**Comment by zepinglee** [comment, marked as answer]
> 目前样式似乎只能插入上标，因此我只能手动在word当中调整格式，但每次新插入文献刷新后，都需要重新调整。

CSL 的引注只能统一选择上标或者非上标格式。

> 请问是否有更简便的解决方式，能够指定该插入的文献为上标或者正常格式。非常感谢！

我建议修改 CSL 的代码去掉 `vertical-align="sup"` 统一设为正文格式，然后在上标中插入引注。

https://github.com/zotero-chinese/styles/blob/33d04873f26f3ede5c5e7633887d8a1b371681b7/src/gb-t-7714-2015-numeric-bilingual/gb-t-7714-2015-numeric-bilingual.csl#L431

**Reply by lsqzero** [reply]
感谢！但如果统一设置为正文格式，是不是还是需要每次刷新以后再手动设置上标？

**Comment by northword** [comment]
我倾向于写成 

人名等¹研究了xxxx。

**Comment by lsqzero** [comment]
各位，最后写（GPT写的）了一段VBA。大致实现了上标和正文文献混排的功能。大致思路是是找到通配符为\[*\]的情况（基本上包含了参考文献，但是数学上闭区间也会被囊括进来，根据需要排除），然后检索前面两三个字符，如果出现了“文献”“与”“和”“=”这样的字符则是正文文本，不是的话则设置成上标。
代码开始：

Sub FormatReferences()
    Dim doc As Document
    Dim rng As Range
    Dim refPattern As String
    Dim refText As String
    Dim prevText As String
    Dim prevWords As Variant
    Dim i As Integer
    Dim found As Boolean
    
    ' 指定要查找的引用模式
    refPattern = "\[*\]"
    
    ' 获取当前文档
    Set doc = ActiveDocument
    
    ' 设置范围为整个文档
    Set rng = doc.Content
    
    ' 使用通配符查找引用
    With rng.Find
        .Text = refPattern
        .MatchWildcards = True
        .Forward = True
        .Wrap = wdFindStop
    End With
    
    ' 遍历找到的引用
    Do While rng.Find.Execute
        ' 获取找到的引用文本
        refText = rng.Text
        
        ' 获取引用前面的1~2个字符
        Set refMatch = rng.Duplicate
        refMatch.MoveStart Unit:=wdCharacter, Count:=-2 ' 向前移动2个字符
        prevText = refMatch.Text
        
        ' 检查前面的文字是否包含“文献”“如”“和”
        prevWords = Array("文献", "如", "和", "=")
        found = False
        For i = LBound(prevWords) To UBound(prevWords)
            If InStr(prevText, prevWords(i)) > 0 Then
                found = True
                Exit For
            End If
        Next i
        
        ' 根据检查结果设置格式
        If found Then
            rng.Font.Superscript = False ' 设置为正文格式
        Else
            rng.Font.Superscript = True ' 设置为上标格式
        End If
    Loop
    
    MsgBox "引用格式设置完成！"
End Sub
